### PR TITLE
Unarmed weapons are wearable and added wrist sub-limbs

### LINKED ISSUE #53822: Briefcase and wrist watch conflict when worn on same side
### Describe the bug

Briefcase and wrist watch conflict when worn on same hand.

### Steps to reproduce

1. Get or cheat yourself a briefcase and a watch (should work with any; only tested with ordinary wrist watch)
2. Press your Sort Armor key
3. Notice how it tells you that "Wearing multiple items strapped to you on your left hand is adding encumbrance there."

### Expected behavior

Shouldn't have a conflict because wrist watches are worn on, supposedly, the wrist. Briefcases are however held in one's hands. At 8 INT, I should be aware of how to wear a watch.

### Screenshots

grafik

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: f8734ab [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    No Monsters [Only_Wildlife],
    Extra Mutated Scenarios [extra_mut_scens],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    SpeedyDex [speedydex]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Hymore246**
#### Summary
Balance "Unarmed weapons are wearable and added wrist sub-limbs"

#### Purpose of change

Fixes #53822
Unarmed weapons are wearable. Added Wrist and Fingers sub-limbs

#### Describe the solution

This PR is focused on several changes regarding the character's "hand" limb. 

**Unarmed Weapons are now wearable.** Following the example of #41209 all other non-CBM unarmed weapons have been converted to `ARMOR` items that are worn on the hands. These items no longer have the `UNARMED_WEAPON` flag meaning they must be worn to add their damage to unarmed attacks. The follow items have been changed:

Name | Encumbrance | Coverage 
--- | --- | --- 
pair of brass knuckles | 5 | 10
pair of nail knuckles | 7 | 20
pair of scrap knuckles | 7 | 20
pair of steel knuckles | 5 | 10

In addition, these items are no longer classified as style weapons for Aikido and Judo. Wearing the above items and attacking without a weapon will still make them contribute unarmed damage as normal. Due to the nature of tiger claws (bagh nakha) and bionic claws, they are still classified as style weapons for Aikido and Judo and retain the `UNARMED_WEAPON` flag.

**Added wrist sub-limbs.** `arm_wrist_r` and `arm_wrist_l` have been added to the body parts list as a sub-limbs of `hand_r` and `hand_l` respectively. Items that cover the wrist have been updated to state they specifically cover the wrist part of the hand. These items are all watches:

* radiation biomonitor
* fitness band
* All watches

**Added fingers sub-limbs.** `arm_fingers_r` and `arm_fingers_l` have been added to the body parts list as a sub-limbs of `hand_r` and `hand_l` respectively. The above unarmed weapons and all ring jewelry use this slot.

### CODE REVIEW

**Comment by bombasticSlacks** [comment]
if something doesn't specify sublocations it is assumed to cover all sublocations on that body part.

So if the intent is for there to be two distinct zones "fingers" where you have brass knuckles or whatever and "wrist" where you have a watch. you have to make a fingers sublocation too.

**Comment by bombasticSlacks** [comment]
also the overall limb needs to have a list of sub parts too. See line 307 for an example
image

**Comment by Shodan14** [comment]
There's also the fitness band and bracelets.

**Comment by NetSysFire** [comment]
Relevant: #53822

**Comment by catdach** [comment]
It seems that in the comments of #41209 it was decided not to have the "knuckles" items be wearable as they actually [take up space in the palm](https://upload.wikimedia.org/wikipedia/commons/1/1d/Brass_knuckles.jpg) and therefore prevent other things from being held in the hands. They don't just wrap around the fingers like the cestus.

**Comment by catdach** [comment]
> > It seems that in the comments of #41209 it was decided not to have the "knuckles" items be wearable as they actually [take up space in the palm](https://upload.wikimedia.org/wikipedia/commons/1/1d/Brass_knuckles.jpg) and therefore prevent other things from being held in the hands. They don't just wrap around the fingers like the cestus.
> 
> I see where that is coming from but there is no mistake that Brass Knuckles and the like need to be placed on the fingers to be used. You couldn't just hold a pair of Spiked Brass Knuckles in your hand and attack with them the same way as if they were worn. The best idea I can think of as a compromise is to make these weapon have a large amount of hand encumbrance to discourage holding another item.

But increasing hand encumbrance would also make unarmed attacks worse too, right?

File: data/json/body_parts.json [hunk 1]
```diff
@@ -615,5 +617,45 @@
...
+    "parent": "hand_r",
+    "side": 1,
+    "opposite": "hand_wrist_l",
+    "name_multiple": "wrists",
+    "name": "right wrist"
+  },
+  {
+    "id": "hand_fingers_r",
+    "type": "sub_body_part",
+    "max_coverage": 5,
```
Comment (bombasticSlacks) [inline data/json/body_parts.json:634]:
```suggestion
    "max_coverage": 50,
```

File: data/json/body_parts.json [hunk 2]
```diff
@@ -615,5 +617,45 @@
...
+    "parent": "hand_r",
+    "side": 1,
+    "opposite": "hand_fingers_l",
+    "name_multiple": "wrists",
+    "name": "right fingers"
+  },
+  {
+    "id": "hand_wrist_l",
+    "type": "sub_body_part",
+    "max_coverage": 5,
```
Comment (bombasticSlacks) [inline data/json/body_parts.json:644]:
```suggestion
    "max_coverage": 10,
```

File: data/json/body_parts.json [hunk 3]
```diff
@@ -615,5 +617,45 @@
...
+  {
+    "id": "hand_fingers_l",
+    "type": "sub_body_part",
+    "max_coverage": 5,
+    "parent": "hand_l",
+    "side": 1,
+    "opposite": "hand_fingers_r",
+    "name_multiple": "fingers",
+    "name": "left fingers"
   }
```
Comment (bombasticSlacks) [inline data/json/body_parts.json:660]:
```suggestion
  },
    {
    "id": "hand_palm_l",
    "type": "sub_body_part",
    "max_coverage": 40,
    "parent": "hand_l",
    "side": 1,
    "opposite": "hand_palm_l",
    "name_multiple": "palms",
    "name": "left palm"
  }
```

File: data/json/items/armor/jewelry.json [hunk 4]
```diff
@@ -231,7 +239,7 @@
     "sided": true,
     "use_action": [ "WEATHER_TOOL" ],
     "flags": [ "WATCH", "ALARMCLOCK", "WATER_FRIENDLY", "THERMOMETER", "BELTED", "ALLOWS_NATURAL_ATTACKS", "OVERSIZE" ],
-    "armor": [ { "coverage": 5, "covers": [ "hand_l", "hand_r" ] } ]
+    "armor": [ { "coverage": 5, "covers": [ "hand_l", "hand_r" ], "specifically_covers": [ "hand_wrist_l", "hand_wrist_r" ] } ]
```
Comment (bombasticSlacks) [inline data/json/items/armor/jewelry.json:242]:
```suggestion
    "armor": [ { "coverage": 50, "covers": [ "hand_l", "hand_r" ], "specifically_covers": [ "hand_wrist_l", "hand_wrist_r" ] } ]
```

File: data/json/items/armor/jewelry.json [hunk 5]
```diff
@@ -924,7 +958,7 @@
     "color": "light_gray",
     "sided": true,
     "flags": [ "WATCH", "ALARMCLOCK", "FANCY", "BELTED", "FRAGILE", "ALLOWS_NATURAL_ATTACKS", "WATER_FRIENDLY", "OVERSIZE" ],
-    "armor": [ { "coverage": 10, "covers": [ "hand_l", "hand_r" ] } ]
+    "armor": [ { "coverage": 5, "covers": [ "hand_l", "hand_r" ], "specifically_covers": [ "hand_wrist_l", "hand_wrist_r" ] } ]
```
Comment (bombasticSlacks) [inline data/json/items/armor/jewelry.json:961]:
```suggestion
    "armor": [ { "coverage": 100, "covers": [ "hand_l", "hand_r" ], "specifically_covers": [ "hand_wrist_l", "hand_wrist_r" ] } ]
```

File: data/json/items/armor/jewelry.json [hunk 6]
```diff
@@ -970,7 +1004,7 @@
     "color": "yellow",
     "sided": true,
     "flags": [ "WATCH", "ALARMCLOCK", "SUPER_FANCY", "BELTED", "FRAGILE", "ALLOWS_NATURAL_ATTACKS", "WATER_FRIENDLY", "OVERSIZE" ],
-    "armor": [ { "coverage": 10, "covers": [ "hand_l", "hand_r" ] } ]
+    "armor": [ { "coverage": 10, "covers": [ "hand_l", "hand_r" ], "specifically_covers": [ "hand_wrist_l", "hand_wrist_r" ] } ]
```
Comment (bombasticSlacks) [inline data/json/items/armor/jewelry.json:1007]:
```suggestion
    "armor": [ { "coverage": 100, "covers": [ "hand_l", "hand_r" ], "specifically_covers": [ "hand_wrist_l", "hand_wrist_r" ] } ]
```

File: data/json/items/melee/unarmed_weapons.json [hunk 7]
```diff
@@ -11,73 +11,112 @@
...
     "bashing": 4,
     "price_postapoc": 180,
     "qualities": [ [ "HAMMER", 1 ] ],
-    "flags": [ "UNARMED_WEAPON", "DURABLE_MELEE" ]
+    "flags": [ "DURABLE_MELEE" ],
+    "material_thickness": 2,
+    "armor": [
+      {
+        "encumbrance": 5,
+        "coverage": 10,
```
Comment (bombasticSlacks) [inline data/json/items/melee/unarmed_weapons.json:115]:
```suggestion
        "coverage": 20,
```

File: data/json/items/melee/unarmed_weapons.json [hunk 8]
```diff
@@ -115,7 +154,14 @@
     "warmth": 0,
     "material_thickness": 3,
     "flags": [ "WATER_FRIENDLY", "DURABLE_MELEE", "OUTER" ],
-    "armor": [ { "encumbrance": 5, "coverage": 40, "covers": [ "hand_l", "hand_r" ] } ]
+    "armor": [
+      {
+        "encumbrance": 5,
+        "coverage": 40,
```
Comment (bombasticSlacks) [inline data/json/items/melee/unarmed_weapons.json:160]:
```suggestion
        "coverage": 80,
```

File: data/json/items/tool_armor.json [hunk 9]
```diff
@@ -2121,7 +2121,7 @@
         "default_magazine": "light_minus_battery_cell"
       }
     ],
-    "armor": [ { "coverage": 5, "covers": [ "hand_l", "hand_r" ] } ]
+    "armor": [ { "coverage": 5, "covers": [ "hand_l", "hand_r" ], "specifically_covers": [ "hand_wrist_l", "hand_wrist_r" ] } ]
```
Comment (bombasticSlacks) [inline data/json/items/tool_armor.json:2124]:
```suggestion
    "armor": [ { "coverage": 50, "covers": [ "hand_l", "hand_r" ], "specifically_covers": [ "hand_wrist_l", "hand_wrist_r" ] } ]
```

**Review by bombasticSlacks — CHANGES_REQUESTED** [review]
Alright so this makes the wrist 10% of the hand, fingers 50%, and adds a palm to round it out covering 40%.

When defining coverage for a place with sublocations you are telling the game *how much of the sublocation is covered* not *how much of the overall location*. So for example when a watch is 10 coverage, on the wrist the game thinks its 10% of the wrist (which is 10% of the hand). So I've adjusted the coverage values to maintain the numbers you already had but in that system where it converts.

File: data/json/body_parts.json [hunk 10]
```diff
@@ -615,5 +617,85 @@
...
+    "side": 1,
+    "opposite": "hand_wrist_r",
+    "name_multiple": "wrists",
+    "name": "left wrist"
+  },
+  {
+    "id": "hand_palm_l",
+    "type": "sub_body_part",
+    "max_coverage": 20,
+    "parent": "hand_r",
```
Comment (bombasticSlacks) [inline data/json/body_parts.json:675]:
```suggestion
    "parent": "hand_l",
```

File: data/json/body_parts.json [hunk 11]
```diff
@@ -615,5 +617,85 @@
...
+    "side": 1,
+    "opposite": "hand_palm_r",
+    "name_multiple": "palms",
+    "name": "left palm"
+  },
+  {
+    "id": "hand_back_l",
+    "type": "sub_body_part",
+    "max_coverage": 20,
+    "parent": "hand_r",
```
Comment (bombasticSlacks) [inline data/json/body_parts.json:685]:
```suggestion
    "parent": "hand_l",
```

**Review by bombasticSlacks — CHANGES_REQUESTED** [review]

**Pull request merged**